lnrpc/routerrpc/router.proto

```diff
@@ -377,6 +377,73 @@ message PairHistory {
     bool last_attempt_successful = 6 [json_name = "last_attempt_successful"];
 }
 
+
+/// HopAddress contains either a channel or a pubkey to address the next hop.
+message HopAddress {
+    oneof type {
+        /// Short channel id.
+        uint64 channel = 1;
+
+        /// 33-byte node pubkey.
+        bytes pubkey = 2;
+    }
+}
+
+/// Defines the various modes BuildRoute can operate in.
+enum BuildRouteMode {
+    /**
+    Build the route for the given amount.
+    */
+    AMT = 0;
+
+    /**
+    Build the route for the minimum amount that the route can carry. Ignore the
+    amt parameter
+    */
+    MIN_AMT = 1;
+
+    /**
+    Select and verify channels for the given amount, but build the final route
```

Comment:
Can't this be simulated by calling with AMT first, then follow up with MIN_AMT?

Reply:
Looking at it again, I found this mode over-complicated. It would only have a function when there would be distinct policies within the set of channels between two nodes. This is discouraged anyway in the spec.